# day08 복습

## 혼돈한 개념1

## read.me 가 있는 repositories를 생성 후 오너가 dev branch 만든 후 push 를 할때 로컬에 있는 파일에서 작업한 후 push 하는 줄 알고 오류가 발생하지 않을까?(initial commit 과 충돌 )

> 래포지토리 생성 > READ.ME 파일 생성
>
> 1.  로컬에다 CLONE -> 로컬에 READ.ME 파일 만듬-> PUSH했을떄 오류 발생X
>
> 2.  로컬에서 INIT --> read.me 파일 없음-> initial commit 과 히스토리가 달라서 > push할경우 오류 발생 > sol1 외부저장소로부터 git pull(동기화) 한다음에 작성--> 히스토리가 같아 오류발생안함 > sol2 git push --> 오류 발생 --> rebase(merge)

---

## 혼돈한 개념2

## dev 에서 개발을 한 후 main에 병합 하는 줄 알았음 --> dev를 commit push 함

> main 은 현재 진행형? branch 이므로 dev라는 branch를 만들고 dev branch 안에서 필요한 기능에 따라 새로운 branch를 생성 -> 수정 후 !새로운 branch!에서 dev branch로
> git add. commit pull request(new branch-> dev) 를 보내기

---

## 혼돈한 개념 3

feat/ signup을 원격 저장소에 push를 할떄 conflict 가 발생하는 줄 알았음.

---

conflict 를 발생하게 되는 이유

> dev에서 style/signup branch를 생성후 feat/signup branch 를 만들어 수정을 하는 도중에 dev branch를 style/signup을 pull 받아 fixed/signup branch 를 만들어 수정하는 작업을 하고 만약 fixed를 pull request를 통해 먼저 merge를 했다면 최종적으로 signup을 merge 하는 과정에서 충돌할 수 있는 상황은

1. feat / signup branch에서 원격 저장소에서 git pull (동기화)힌 후 merge 할 떄--> commit 내용이 다르다.

2. 로컬에 있는 dev 브렌치에 원격 저장소에서 git pull을 받아 dev branch에서 merge 할떄 --> 동일
